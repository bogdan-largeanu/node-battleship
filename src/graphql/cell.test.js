describe('GraphQL Type: Cell', () => {
    it(`fetch cell's only scalar values`, async () => {
        const { data } = await query(
            {
                query: `
                {
                    cell(id: 1) {
                        id
                        seq
                        coordinate
                    }
                }`
            }
        );

        expect(data).toMatchSnapshot();
    });

    it(`fetch cell's battlefield's scalar values`, async () => {
        const { data } = await query(
            {
                query: `
                {
                    cell(id: 1) {
                        battlefield {
                            id
                        }
                    }
                }`
            }
        );

        expect(data).toMatchSnapshot();
    });
});
