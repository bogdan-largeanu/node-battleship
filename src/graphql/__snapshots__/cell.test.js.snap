// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQL Type: Cell fetch cell's battlefield's scalar values 1`] = `
Object {
  "cell": Object {
    "battlefield": Object {
      "id": "2",
    },
  },
}
`;

exports[`GraphQL Type: Cell fetch cell's only scalar values 1`] = `
Object {
  "cell": Object {
    "coordinate": "A1",
    "id": "1",
    "seq": 0,
  },
}
`;
